docs/source/onnx.rst

```diff
@@ -79,433 +84,154 @@ Then, you can run::
-dictionary as the last input in the tuple args in such cases.
-The new call would look like this. ::
+Using the Tensor.data field can produce an incorrect trace and therefore an incorrect ONNX graph.
+Use :func:`torch.Tensor.detach` instead. (Work is ongoing to
+`remove Tensor.data entirely <https://github.com/pytorch/pytorch/issues/30987>`_).
 
-    torch.onnx.export(model, (k, x, {}), ‘test.onnx’)
 
+Limitations
+-----------
 
-Indexing
---------
-
-Tensor indexing in PyTorch is very flexible and complicated.
-There are two categories of indexing. Both are largely supported in exporting today.
-If you are experiencing issues exporting indexing that belongs to the supported patterns below,
-please double check that you are exporting with the latest opset (opset_version=12).
-
-Getter
-~~~~~~
-
-This type of indexing occurs on the RHS. Export is supported for ONNX opset version >= 9. E.g.: ::
-
-  data = torch.randn(3, 4)
-  index = torch.tensor([1, 2])
-
-  # RHS indexing is supported in ONNX opset >= 11.
-  class RHSIndexing(torch.nn.Module):
-      def forward(self, data, index):
-          return data[index]
-
-  out = RHSIndexing()(data, index)
-
-  torch.onnx.export(RHSIndexing(), (data, index), 'indexing.onnx', opset_version=9)
+Types
+^^^^^
 
-  # onnxruntime
-  import onnxruntime
-  sess = onnxruntime.InferenceSession('indexing.onnx')
-  out_ort = sess.run(None, {
-      sess.get_inputs()[0].name: data.numpy(),
-      sess.get_inputs()[1].name: index.numpy(),
-  })
+* Only torch.Tensors, numeric types that can be trivially converted to torch.Tensors (e.g. float, int),
+  and tuples and lists of those types are supported as model inputs or outputs. Dict and str inputs and
+  outputs are accepted in :ref:`tracing<tracing-vs-scripting>` mode, but:
 
-  assert torch.all(torch.eq(out, torch.tensor(out_ort)))
+  * Any computation that depends on the value of a dict or a str input will be replaced with the
+    constant value seen during the one traced execution.
+  * Any output that is a dict will be silently replaced with a flattened sequence of its values
+    (keys will be removed). E.g. ``{"foo": 1, "bar": 2}`` becomes ``(1, 2)``.
+  * Any output that is a str will be silently removed.
 
-Below is the list of supported patterns for RHS indexing. ::
+* Certain operations involving tuples and lists are not supported in
+  :ref:`scripting<tracing-vs-scripting>` mode due to limited support in ONNX for nested sequences.
+  In particular appending a tuple to a list is not supported. In tracing mode, the nested sequences
+  will be flattened automatically during the tracing.
 
-  # Scalar indices
-  data[0, 1]
+Differences in Operator Implementations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-  # Slice indices
-  data[:3]
+Due to differences in implementations of operators, running the exported model on different runtimes
+may produce different results from each other or from PyTorch.
```

Comment:
The difference need be in a small threshold, since we need expect the results are consistent.

Reply:
Tried to clarify.